File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.cs
public class Solution {
    public bool CanMakeSubsequence(string str1, string str2) {
        
        if(str2.Length > str1.Length)return false;
        
        int indx = 0;

        for(int i = 0 ; i < str1.Length ; i++){

            if(indx >= str2.Length) break;

            if(str1[i] == str2[indx])indx++;

            else if(str1[i] == 'z' && str2[indx] == 'a')indx++;

            else if(str1[i] + 1 == str2[indx])indx++;
            
        }
        if(indx >= str2.Length) return true;
        return false;
    }
}

File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        if(str2.size() > str1.size()) return false;
        int indx = 0;
        bool ok = 0;
        for(int i = 0 ; i < str1.size() ; i++){
            if(indx >= str2.size()) break;
            if(str1[i] == str2[indx])indx++;
            else if(str1[i] == 'z' && str2[indx] == 'a')indx++;
            else if(str1[i]+1 == str2[indx])indx++;

        }
        if(indx >= str2.size() )return true;
        return false;
    }
};